WordPress/src/main/java/org/wordpress/android/ui/activitylog/BackupDownloadProgressChecker.kt

```diff
@@ -0,0 +1,61 @@
+package org.wordpress.android.ui.activitylog
+
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.isActive
+import kotlinx.coroutines.withContext
+import org.wordpress.android.fluxc.action.ActivityLogAction.FETCH_BACKUP_DOWNLOAD_STATE
+import org.wordpress.android.fluxc.model.SiteModel
+import org.wordpress.android.fluxc.store.ActivityLogStore
+import org.wordpress.android.fluxc.store.ActivityLogStore.FetchBackupDownloadStatePayload
+import org.wordpress.android.fluxc.store.ActivityLogStore.OnBackupDownloadStatusFetched
+import org.wordpress.android.modules.DEFAULT_SCOPE
+import javax.inject.Inject
+import javax.inject.Named
+import javax.inject.Singleton
+
+@Singleton
+class BackupDownloadProgressChecker
+@Inject constructor(
+    private val activityLogStore: ActivityLogStore,
+    @param:Named(DEFAULT_SCOPE) private val defaultScope: CoroutineScope
+) {
+    companion object {
```

Comment:
Nitpick: Again something which is consistent with the RewindProgressChecker but AFAIK not with the rest of the codebase. We usually create constants as private vals above the class, not in companion objects.

Reply:
I'll tidy that up. 👍